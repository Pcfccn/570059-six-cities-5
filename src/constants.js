const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  OPEN_SORT_OPTIONS: `OPEN_SORT_OPTIONS`,
  CHANGE_SORT_TYPE: `OPEN_SORT_TYPE`,
  CHANGE_ENTERED_OFFER: `CHANGE_ENTERED_OFFER`,
  LOAD_OFFERS: `LOAD_OFFERS`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
  LOAD_USER_DATA: `LOAD_USER_DATA`,
  LOAD_ONE_OFFER: `LOAD_ONE_OFFER`,
  LOAD_COMMENTS: `LOAD_COMMENTS`,
  LOAD_FAVORITES: `LOAD_FAVORITES`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
  SET_RATING: `SET_RATING`,
  ENTER_TEXT: `ENTER_TEXT`,
  CHANGE_COMMENT_SENDING_STATUS: `CHANGE_COMMENT_SENDING_STATUS`,
};

const Path = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  getUrlById: (id) => (`/offer/${id}`)
};
const ApiURL = {
  LOGIN: `/login`,
  HOTELS: `/hotels`,
  FAVORITES: `/favorite`,
  getUrlById: (id) => (`/hotels/${id}`),
  getUrlNearbyById: (id) => (`/hotels/${id}/nearby`),
  getUrlCommentsById: (id) => (`/comments/${id}`),
  getUrlFavoriteStatus: (id, status) => (`/favorite/${id}/${status}`),
};

const ratingStars = [5, 4, 3, 2, 1];
const ratingInputTitle = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

const widthStartsCoefficient = 20;

const OfferCardClassName = {
  NEAR_PLACE: `near-places__card`,
  CITIES_PLACE: `cities__place-card`,
};

const MapClassName = {
  PROPERTY: `property__map`,
  CITIES: `cities__map`,
};

const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

const SortType = {
  PRICE_ASC: `Price: low to high`,
  PRICE_DESC: `Price: high to low`,
  POPULAR_DESC: `Popular`,
  RATING_DESC: `Top rated first`,
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const BookmarksButtonType = {
  PROPERTY: `PROPERTY`,
  PLACE_CARD: `PLACE_CARD`,
};

const CommentSendingStatus = {
  SENDING: `SENDING`,
  DONE: `DONE`,
  ERROR: `ERROR`,
};

export {ActionType, Path, ApiURL, widthStartsCoefficient, ratingStars, ratingInputTitle, OfferCardClassName,
  MapClassName, cities, SortType, AuthorizationStatus, BookmarksButtonType, CommentSendingStatus};
