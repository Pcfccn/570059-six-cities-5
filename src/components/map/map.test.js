import {createBrowserHistory} from "history";
import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import {MapClassName} from "../../constants";
import MapComponent from "./map";

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        id: index + 1,
        city: `Amsterdam`,
        reviews: [index + 2134, index + 2135],
        image: [
          `/img/apartment-01.jpg`,
          `/img/apartment-02.jpg`,
          `/img/apartment-03.jpg`,
          `/img/room.jpg`,
          `/img/studio-01.jpg`,
          `/img/studio-photos.jpg`,
        ],
        isPremium: Boolean(Math.random()),
        isInBookmarks: Boolean(Math.random()),
        name: `Beautiful & luxurious studio`,
        type: `Apartament`,
        bedrooms: 3,
        adults: 4,
        rating: 3.5,
        price: {
          value: 1234,
          period: `night`,
        },
        inside: [
          `Wi-Fi`,
          `Washing machine`,
          `Towels`,
          `Heating`,
          `Coffee machine`,
          `Baby seat`,
          `Kitchen`,
          `Dishwasher`,
          `whatewer you want`,
        ],
        host: {
          avatar: `/img/avatar-angelina.jpg`,
          name: `Anagelina`,
          pro: Boolean(Math.random()),
          description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
        },
        location: [52.3909553943508, 4.85309666406198],
        previewImage: `/img/avatar-angelina.jpg`,
        cityLocation: [52.369553943508, 4.85309666406198],
        cityZoom: 10,
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};

it(`Should Map Component render correctly`, () => {
  const offers = getOffersMock(4);
  const enteredOfferLocation = getOffersMock(1).cityLocation;
  const history = createBrowserHistory();

  const tree = renderer
    .create(
        <Router history={history}>
          <MapComponent
            key={`key`}
            className={MapClassName.CITIES}
            cityLocation={offers[0].cityLocation}
            zoom={offers[0].cityZoom}
            pinLocations={offers
            .map((offer) => offer.location)
            .filter((offerLocation) => offerLocation !== enteredOfferLocation)}
            chosedPinLocation={enteredOfferLocation}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
