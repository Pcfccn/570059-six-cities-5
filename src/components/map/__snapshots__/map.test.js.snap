// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Map Component render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MapComponent
    chosedPinLocation={
      Array [
        52.369553943508,
        4.85309666406198,
      ]
    }
    cityLocation={
      Array [
        52.369553943508,
        4.85309666406198,
      ]
    }
    className="cities__map"
    key="key"
    pinLocations={
      Array [
        Array [
          52.3909553943508,
          4.85309666406198,
        ],
        Array [
          52.3909553943508,
          4.85309666406198,
        ],
        Array [
          52.3909553943508,
          4.85309666406198,
        ],
        Array [
          52.3909553943508,
          4.85309666406198,
        ],
      ]
    }
    zoom={10}
  >
    <section
      className="cities__map map"
      id="map"
    />
  </MapComponent>
</Router>
`;
